src/Product/Single.php

```diff
@@ -129,8 +145,28 @@ public function init_hooks() {
 		add_action( 'woocommerce_process_product_meta', array( $this, 'save_additional_product_parent_options' ) );
 		add_action( 'woocommerce_variation_options_pricing', array( $this, 'additional_product_variation_shipping_options' ), 10, 3 );
 		add_action( 'woocommerce_save_product_variation', array( $this, 'save_additional_product_variation_options' ), 10, 2 );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_product_single_css_script' ) );
+		add_action( 'woocommerce_before_product_object_save', array( $this, 'validate_product_before_save' ), 100, 1 );
+
 	}
 
+	/**
+	 * Enqueue JS file in product single page.
+	 */
+	public function enqueue_product_single_css_script() {
```

Comment:
The current function name `enqueue_product_single_css_script()` is a bit misleading
```suggestion
	public function enqueue_admin_product_edit_script() {
```